include/nil/crypto3/zk/commitments/detail/polynomial/basic_fri.hpp

```diff
@@ -681,10 +684,16 @@ namespace nil {
                         math::polynomial_dfs<typename FRI::field_type::value_type>,
                         PolynomialType>::value
                     ) {
+                        std::unordered_map<std::size_t,
+                                           std::shared_ptr<math::evaluation_domain<typename FRI::field_type>>> d_cache;
                         for (const auto &[key, poly_vector]: g) {
                             for (const auto& poly: poly_vector) {
                                 if (poly.size() != fri_params.D[0]->size()) {
-                                   g_coeffs[key].emplace_back(poly.coefficients());
+                                    if (d_cache.find(poly.size()) == d_cache.end()) { [[unlikely]]
```

Comment:
[[unlikely]]???

Reply:
Most of the time we should have a cache hit, and this should help with branch prediction (at least according to clang/gcc docs, no idea how they actually control the predictor's behavior).